### PR TITLE
Tweak the page title for code intelligence settings.

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, the page title was "404 not found" while the code
intelligence settings UI was loading in the browser. The title would
then get updated to the correct title after the SPA finished loading.
This commit adds a server-side route handler so that the title is "Code
intelligence" instead of "404 not found" while the page is loading.

We also remove the "Precise" prefix from the title in code intel
settings pages.  The "Precise" prefix is technically accurate, but it's
inconsistent with the "Code Intelligence" button that users click to view
those settings. "Precise code intelligence" is also so long that it's
usually truncated by the browser so that you can't see the
config/uploads/auto-index suffix in the subpages.

CleanShot 2021-08-03 at 10 03 31

### CODE REVIEW

File: cmd/frontend/internal/app/ui/router.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ const (
 	routeSearchBadge      = "search-badge"
 	routeRepo             = "repo"
 	routeRepoSettings     = "repo-settings"
+	routeCodeIntelligence = "code-intelligence"
```
Comment (eseliger) [inline cmd/frontend/internal/app/ui/router.go:39]: To be consistent

```suggestion
	routeRepoCodeIntelligence = "repo-code-intelligence"
```

**Review by eseliger — APPROVED** [review]
🌟

File: cmd/frontend/internal/app/ui/router.go (hunk 1 above)
Comment (olafurpg) [reply cmd/frontend/internal/app/ui/router.go:39]: Good suggestion 👍

**Review by efritz — APPROVED** [review]
Had no clue about the router thing!

**Review by eseliger — APPROVED** [review]

**Comment by eseliger** [comment]
The build is failing

**Comment by olafurpg** [comment]
@eseliger I've been meaning to resolve the build failures for a while but I wasn't able to reproduce them locally. I just rebased for now, will try to get this PR green later this week :)

**Comment by eseliger** [comment]
I think it'll pass now :)

**Comment by olafurpg** [comment]
@eseliger It did! :))

**Pull request merged**